### DISCUSSION TITLE
Video not downloading | Maestro

### DISCUSSION THREAD

**Discussion by joseHbernardino** [aclap-dev/video-downloadhelper#1416] [Help and Q&A]
### Your question

I am unable to download the video using the extension. Checking the video details, it returns the following data:

Page URL: https://app.pato.academy/play/a4940.....
video component: https://maestro-v3.pato.academy/uZQe6qYbez6.....
core media: Mp4 b:HTTPMedia p:non-adaptative unknowns
video: unknown at -1, fps: ?, q:?, no dim

![image](https://github.com/user-attachments/assets/123e3a8f-c26c-4b0b-9471-c6bcfeba03e5)

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: pt-BR
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"6f7784b7a0c74be4"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:131.0) Gecko/20100101 Firefox/131.0
{
    "valid_license_message_has_been_displayed": true,
    "theme": "system",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": true,
    "open_count_store": 2
}

**Comment by Mattmjs** [comment]
This is a video that can be downloaded with either the browser or the companion app. However sometimes it will also fail using
the browser download option. In that case using the companion app should help. Your details provided doesn't show which
download method is used.

**Comment by joseHbernardino** [comment]
How can I download using the companion app? 

And how can I get the information to know if the video can be downloaded or not.

**Comment by Mattmjs** [comment]
This setting controls if browser or companion app is used. See the image. Make sure the box is not ticked.
![VDH settings](https://github.com/user-attachments/assets/a78c7d66-637b-4e66-bf1e-f93225a37ca2)

**Comment by joseHbernardino** [comment]
![image](https://github.com/user-attachments/assets/ff41626d-4306-421d-a5b4-7208c6a6844b)


The option is not selected.

And when I refresh the page this is what I did appearing. It does not detect the entire video to download.

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

What happens when you go ahead & try to download any of those variants?  It is possible that despite the indications of short duration, that VDH will actually download something much longer than what it shows there.  It's also possible that you're going to have to try to download more than one of those things, perhaps all of them, before you find the one you're actually looking for.  Unfortunately, this appears to be a site that requires a registration, so you're the only one who can investigate what's going on.  I'm not going to register for the site.  Also, the second URL you posted in your original post is now giving 404 Not Found.

**Reply by GrampaWildWilly** [reply]
I would also recommend you change your default action to `Download As` instead of `Download`.  That way, you will be able to name each downloaded file yourself instead of taking the default provided by VDH.  There are discussions in this forum on the subject of the default action.  You need to search for them & read them.

**Reply by Mattmjs** [reply]
@joseHbernardino maybe this is not a http
media like your original post suggests going by
your screenshot. Since you're using Firefox can
you verify this shown in the image. This will tell
you if it's protected with DRM.

![drm](https://github.com/user-attachments/assets/ac080353-3da9-416a-ae57-ca746350453e)

**Comment by joseHbernardino** [comment]
I tried to download several files. But they all keep failing.
And when the video is playing, more files appear to be downloaded. But all the same with 0 minutes and HTTPMedia and RawHIS.

![image](https://github.com/user-attachments/assets/a01b8b93-78f3-4e73-880e-3596c97441ce)


![image](https://github.com/user-attachments/assets/91812b25-3947-4ff6-a197-2f8d91bf67ef)

**Reply by Mattmjs** [reply]
I see there is no icon in your image, which means DRM can be ruled out. But there is nothing showing a duration of your video in
the VDH menu. How long is the video ?

You need to look for something matching the duration and try downloading it. Ignore any other options you see.

**Comment by joseHbernardino** [comment]
This video mentioned lasts 13 minutes. But any file that is detected it gets some time. 
Explaining better: all the files that the download helper detects all come with a duration of 0 seconds or without information about seconds in it.

**Reply by Mattmjs** [reply]
Ok I think there is some protection used on this
video. I wouldn't know unless I could view it
myself but I don't have access as it requires a 
login. 

You might need to share the network
activity .

To do that go to your video again and make sure the page has stopped loading, then open the developer tools : Ctrl + Shift + E.
This should open it with the Network selected. Reload the page as well as play the video for about 5 seconds.

After that follow the image to save it. Attach it here as .zip


![374948631-c77203a4-a169-404a-b5a1-99c5a6fc6bcf](https://github.com/user-attachments/assets/dfab6981-528f-4a7d-8ae0-1de35a36fe9e)

**Comment by joseHbernardino** [comment]
[Uploading app.pato.academy_Archive [24-11-03 22-05-49].zip…]()

**Reply by GrampaWildWilly** [reply]
There's something wrong with that link.  When I click it, it just displays this web page again.  You should try to attach that zip file to a post here again.  DO NOT USE E-MAIL for this.  You can't post attachments via E-mail.  That's a GitHub restriction, not an issue with this particular forum.  You have to log on here with your browser & use the button below the text compose box in this web page to attach files.

**Reply by joseHbernardino** [reply]
[app.pato.academy_Archive [24-11-03 22-05-49].zip](https://github.com/user-attachments/files/17613304/app.pato.academy_Archive.24-11-03.22-05-49.zip)

**Reply by Mattmjs** [reply]
I was hoping this would provide some insight why it doesn't work , it still doesn't help much. I found the stream manifest for the
video but it doesn't allow it to be opened outside of the site. I tried looking at the contents of the stream manifest but unreadable.
You'll see how it looks below. I tried to use ffmpeg on the stream manifest but returned an error.
My best guess is that it's using encryption which means there is no way around that.

![Screenshot 2024-11-04 154428](https://github.com/user-attachments/assets/9380acd4-94f1-4a0e-8fa7-ec540c210c90)

Contents of the stream manifest.
`ᢢ䐦䁴ୀ春⠶H⥀ⴸ栺䐢⠹⁄䂙ØǓᘸ$Ǥʸϒ〳䁣N☡᧰ސӖ瀡㏆᷻䀣䁓岠㵂䂓⣑ÎfæĖÖ唦ٴùᩋ䬹怺槭ڇủᔼ琡屙木ᦆ砡ᢕᑚfᩅؠ˩怣ᩰŸ椌♋✼䦦両ã⁠ĥ㠬ဨ(丷ࡱŀ儧⡱㨙%,僀৆汇戄⅚ⵀ牌ㅙ❝থщ㞱⬨ÜB㡠ƠǪ姶࣒䒶Ꮥ删ᣙࡄီ敽Ⱐ榰殻㑙㡮䣦ᝀ㿯᜔Ƥ撧㡢窀࠳坁₅"♐⸐䊴໱ð䊨ྦ帆斿Ϥ燨帠ȍ扈㈫Ꭳ堡屺♩䓀拷ⅱ䱋ࣰ䀨•稈䄵স枼䰠᪭倢᳠ᘠ໏䇩ᐠૠԖ縂㵘⮥ⳤ玠ࣄ曒⪜ТԼಂՠŨत䠸䑳Ѹࡎ⎱硦㟦⽘ত㫽㔀䵦₳噻⺀杰䝈ᛋŐ在拱ⅵ㈫企㟦ᆨ惌函〢㢲⃁⸕㏢狚㻒׶⏋Ḗ崨ⱈ㆏Ⲁ⃘⚒◳䚊挱3甙⻔ᗮ疺溹䔪嚊戡²Ęᒖᆐ榤Πᄊ籒渲ⱇ亴獙営൛˧ၔ൐䟧መ♜⻖右㑙溻煈㐼⟂ヰ丬߇⳰㣿߃炲炿ཻ碝䓲♔࠺ℸ✨ᅄ⛰兔↩ㅅ䝃沃泀岎瑷ৡ硆ҁ敵核㝹潙ࡒӘ噢ᥛ❹泒ృ֤摲ⓠ瑙෡剮Ҁ瘵ཱྀᡙ⮤ⲁЃ堢႒ࢃ乢⌽ຒ⢓䮑ႛ䫒Ḱ䬓帰ᢐ亐仒䉹὆ᓧ䌠幗䙄Ⓚ䢆欸߄㙅ᝀ扃ᛆ倭᥅㚔斑ⲧ䝁ኛኄ乓䏅擛䶆Ṑ䫄糩ឆ⡗ᠴ㴓万唑ᧆ抸䔴ᓀ坃㓰弶䳣ᔸ˴切単处ዩᘅȴ必ṍ䦷઩ᕅሸই⊻ḳḺ宰䫓Ꮓ卦弄媵ᲅ㋭ᢆ᫆徴嫣Ỷ′帱䉪ே᫧䕅⛇ᗄ琫៵ᛏዖ㫍䫐ᜑ哔䋡孶㓏ᒰ໪帲抽坶଄晕⺬劔㫛叁⸻䣕Ṣ宖糍䃱狪幵૷嚗廫嚷欰㨥㺜ఒ䢟䠪㩁奅䉓ᤖ↡儇ᩬଯ懤ぐ刋弖竏嗶绩᷄ử坁ɛᅭ紝堮∍北⛚垄㛒㲗㈖㰄䉹峬㳘洮嚨婮໺㴖秇ᔬ窬㱌绖㫔䊔愬㲸൷ᖹ嬷䈩㥴恖巏㋰俍勚橌曣Ὄ槦婎㸿䠓䒠⡮仴䀆渚㳌ᷘ㔎䠷㲭巟ᵭ㺸૬旽廔⬓㏭䋝㏃ዯ奌总岎㔍ᩖ渉㜽泷ᒼළㄾ䲡㻌洄妌㨎崼㲸䒍௭ᡷ冺㌱㇘瘆ᐆ祕只㨳⯒瘏毳䘄㼆總揙婾Ṉ慾㒩瑍䯴硟䊁皌氌瑭䊕䥞波瘵ʁ矾Ṅ䗍៍䦽㻟捝ࠗ增㳶瑬䉭㾽劦喞暯㆜ߝ痓乵玅ቑ烾ဉᘷ⾍㣱氱姹ᆪᥣ᜗佃瑾恦䂘栌ǨІ带㳚穙࿗捈ඖ傊㢇栜䟷㷴圦傔כ㰕攺㒪㠈㋧洜Ṳ䄋侙沥檀攧徊ா㨵湫枅⬤䏅ఛ傖筄ᘗ漻皁᭏᪏柢彶⓿䯪⹌䜥㘬ᄗ烡㨶⇜ᰣᕲ烿孨᪢壹㲀⢽ԁ᪎㡜ᡣ剜嗴⹢羻筜⎩壑ਮ崋㼇ࠣ呐撔Ⰸ擅䘤擗ᱤཡἹ児愢ӰḌ㨂䱭䘈乖⃸信碷朎怬㓳ぢ㮙笴䒤᎒終慱ࢢᇕ䑳想϶Ꮵ㒂䙫㫣磍଒䶤煻昢栤Ũ᳀稿⍀ဠ `

**Reply by GrampaWildWilly** [reply]
That second one is a zip archive I was able to download.  But I'm as baffled as Matt.  There is one item in the HAR that is identified as having a type of application/vnd.apple.mpegurl.  That is the correct type for an HLS manifest.  Whether it's master or stream can be determined only by looking inside it.  But the contents of that file are NOT a manifest.  You see, no matter what language may be native to the web site in question, all HTML, manifests, & other objects along those lines must be written using the English alphabet.  Well, that's a bit egotistical.  It's not an alphabet that's exclusive to English.  The same alphabet is used in French, Italian, Spanish, German, Swedish, Norwegian, Danish, Finnish, in fact with only slight variations like accents & various diacriticals, in most European languages.  The site might be in China but its HTML & manifests have to use this alphabet.  The fact that this file does not use this alphabet disqualifies it as a manifest.  The web developers on the site are breaking the rules.  There's little hope VDH will be able to deal with this.

Just for yucks, I pushed the contents of the alleged manifest into Google Translate.  Here's what I got:

`
ᢢ䐦䁴ୀ春⠶H⥀ⴸ栺䐢⠹⁄䂙ØǓᘸ$Ǥʸϒ〳䁣N☡᧰伐Ֆ瀡㏆᷻䀣䁓岠㵂䂓⣑ÎfæĖÖ唦ٴùᩋ䬹怺槭ڇủᔼ琡屙木ᦆ砡ᢕᑚfᩅؠ˩怣ᩰŸ椌♋✼䦦両ã⁠ĥ㠬ဨ(丷ࡱŀ儧⡱㨙%,僀৆戄⅚ⵀbrandㅙ❝থщ㞱⬨ÜB㡠ƠǪ姶࣒䒶Ꮥ Deleteᣙࡄီ Ⱐ榰凯㑙㡮䣦ᝀ㿯᜔Ƥ撧㡢窀࠳坁₅"♐⸐䊴໱ð䊨ྦ风斿Ϥ縨丠ȍ扈㈫Ꭳ FORtress 屺♩䓀Copyⅱ䱋ࣰ䀨 •秆䄵স枼䰠᪭倢᳠ᘠ໏ The total number of玠ࣄ曒⪜ТռಂՠŨत䠸䑳Ѹࡎ⎱㟦⽘ত㫽㔀䵦₳thi⺀je䝈ᛋŐ在archⅵ㈫成㟦ᆨ惌 Letter 〢㢲⃁⸕㏢狚㻒׶⏋ Ḗ娨ⱈ㆏Ⲁ⃘⚒◳䚊挱3甙⻔ᗮ疺溹䔪嚊²Ęᒖᆐ榤Πᄊ籒enderⱇ亴獙営൛˧ၔ൐䟧መ♜⻖right㑙溻煈㐼⟂ヰ丬⳰㣿out炲炿ཻ碝瓲♔࠺ℸ✨ᅄ⛰Rabbit ↩ㅅ両剐Հ瘵ཱྀᡙ⮤ⲁЃ堢႒ࢃ乢⌽ຒ⢓䮑ႛ䫒Ḱ䬓帰ᢐ亐仒䉹὆ᓧ䌠帼䙄Ⓚ丢䬸㙅ᝀᛆ倭᥅㚔片ⲧ䝁ኛኄpong䏅擛䶆Ṑ䫄糩ឆ⡗ᠴ㴓万AZoleᧆ且ᓀ坃㓰伶䳣ᔸ˴ Cut 単ዩ ᘅȴ必ṍ䦷઩ᕅሸই⊻ ḳḺzai 䫓Ꮓ Gua Nong媵ᲅ㋭ᢆ᫆徴嫣Ỷ′帱䉪ே᫧䕅⛇ᗄ琫៵ᛏዖ㫍䫐ᜑBLE䋡孶㓏ᒰ໪串坶଄halo⺬劔㫛品⸻丣Ṣ宖獍䃱狪嵵૷嚗廫嚷㨥㺜ఒ䢟䠪㩁奅䉓ᤖ↡儇ᩬଯ捤ぐ弖竏尰᷄ửɛᅭ纴堮∍北⛚ridge㛒㲗㈖㰄䉹峬㳘殮婮婮໺㴖秇ᔬ窬㱌绖㫔䊔愬㲸൷ᖹ嬷䈩㥴恖巏㋰俚橌曣Ὄ槦婎㸿䠓䒠⡮仴䀆渚㳌ᷘ㔎䠷㲭巟ᵭ㺸૬旽廔⬓㏭䋝㏃ዯ奌华㔍ᩖ渉㜽泷ᒼළㄾ䲡㻌洄妌㨎崼㲸䒍௭ᡷ冺㌱㇘瘆ᐆ secret only㨳⯒瘏毳䘄㼆 total 揙婾Ṉdesire 㒩瑍䯴硟䊁皌氇瑭䊕䥞䘵絵ʁ Alum Increase 㳶瑬䉭㾽劦喞暯㆜chuure妦傔כ㰕攺㒪㠈㋧洜Ṳ䄋侙乌俥㘬ᄗ烡㨶⇜ᰣᕲ烿孨᪢壹㲀⢽criminal ᡣ剜嗴⹢绻筜⎩壑ਮ崋㼇ࠣNa撔Ⰸ好䘤擗ᱤཡἹ児愢հḌ㨂䱭䘈好⃸信碷朎怬㓳ぢ㮙笴䒤᎒finally ࢢᇕ䑳think϶Ꮵ㒂䙫㫣磍଒䶤煻昢栤Ũ᳀draft⍀ ဠ
`

As you can see, here & there it managed to translate a word or two.  For the most part, it didn't recognize this content as language.  Google claimed this is Traditional Chinese.  Maybe here & there you've got some Chinese.  The rest of it is in some language Google doesn't recognize just as I don't.  Even so, the words it did manage to translate are not the sort of thing that you would see in a proper HLS manifest.  Whatever this is, it's not a manifest.

Wait.  Time for a brain wave.  Instead of letting Google Translate automatically detect a language, Traditional Chinese being the one it chose, I overrode the choice & tried Simplified Chinese.  Here's what I got:

`
The 100 most common types of plastic bags are made from plastic bags, which are made from plastic bags made from plastic bags. These bags are made from plastic bags made from plastic bags, which are made from plastic bags made from plastic bags. The 100 most common types of plastic bags are made from plastic bags made from plastic bags, which are made from plastic bags made from plastic bags. ❝In the past, the United States has been in a state of panic, and ... The 1988 Beijing International Airport has been a major port city for 12 years, and the 1988 Beijing International Airport has been a major port for 12 years. In this regard, the development of the company has been continuously improved, and the development of the company has been carried out in an orderly manner. ᗮ疺溹五嚊²Ęᒖᆐ榤Πᄊ籒狠ⱇ亴獙営൛˧ၔ൐䟧መ♜⻖right㑙溻煈㐼⟂ヰ丬⳰㣿out炲炿ཻ碝䓲♔࠺ℸ✨ In fact, the enemy's nuclear weapons have been destroyed, and the enemy's nuclear weapons have been destroyed. In fact, the enemy's nuclear weapons have been destroyed, and the enemy's nuclear weapons have been destroyed. The 1988 World Cup has attracted 15 million viewers, and 15 million viewers have signed an agreement with the European and American governments to jointly develop a 1988 World Cup. The only thing you need to do is to make sure that the product is as good as the original one. The 3D printing industry has been developing rapidly, and the industry has also been developing rapidly. In this regard, the development of the economy has entered a new stage, and the development of the economy has entered a new stage.㸿䠓䒠⡮仴䀆渚㳌ᷘ㔎䠷㲭巟ᵭ㺸૬廔⬓㏭䋝㏃ዯ奌总厎㔍ᩖ渉㜽峷ᒼළㄾ䲡㻌洄妌㨎崼㲸䒍௭ᡷ冺㌱㇘瘆In fact, the only thing you can do is to increase your chances of getting enough exercise. In this regard, the development of the Internet has become a hot topic in the industry, and the development of the Internet has become a hot topic in the industry. The 1988 Beijing International Airport has been operating for 100 years and has been operating for 100 years. We will do our best to ensure that your company will always be as trustworthy as possible and that your company will always be as trustworthy as possible.
`

Aside from being rather repetitive it doesn't make a whole lot of sense.  Plus there's still some strings it failed to translate.  Moreover, I don't see any of the words in this translation from Simplified Chinese that Google generated translating from Traditional Chinese.  In any case, this is most assuredly not an HLS manifest.  The web site is faulty.  I can understand completely why VDH doesn't work here.

To illustrate what I'm talking about, I went to this page:

https://www.medici.tv/en/operas/il-pirata-bellini-teatro-real

The Network Monitor gave me a manifest whose contents were this:

```
#EXTM3U
#EXT-X-VERSION:4
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio-aacl-128",NAME="audio",DEFAULT=YES,AUTOSELECT=YES
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio-aacl-160",NAME="audio",DEFAULT=YES,AUTOSELECT=YES
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio-aacl-256",NAME="audio",DEFAULT=YES,AUTOSELECT=YES
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio-aacl-320",NAME="audio",DEFAULT=YES,AUTOSELECT=YES
#EXT-X-MEDIA:URI="https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/9378-DgQp6dfzKJ4WHMj-drm-aes-textstream_eng=1000.m3u8?vbegin=8502&vend=8537",TYPE=SUBTITLES,GROUP-ID="textstream",LANGUAGE="en",NAME="English",AUTOSELECT=NO
#EXT-X-MEDIA:URI="https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/9378-DgQp6dfzKJ4WHMj-drm-aes-textstream_spa=1000.m3u8?vbegin=8502&vend=8537",TYPE=SUBTITLES,GROUP-ID="textstream",LANGUAGE="es",NAME="Spanish; Castilian",AUTOSELECT=NO
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=243000,RESOLUTION=256x144,CODECS="mp4a.40.2,avc1.4D400C",AUDIO="audio-aacl-128",SUBTITLES="textstream"
https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/9378-DgQp6dfzKJ4WHMj-drm-aes-audio=128000-video=99963.m3u8?vbegin=8502&vend=8537
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=765000,RESOLUTION=426x240,CODECS="mp4a.40.2,avc1.4D4015",AUDIO="audio-aacl-160",SUBTITLES="textstream"
https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/9378-DgQp6dfzKJ4WHMj-drm-aes-audio=160000-video=559836.m3u8?vbegin=8502&vend=8537
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=1195000,RESOLUTION=640x360,CODECS="mp4a.40.2,avc1.4D401E",AUDIO="audio-aacl-256",SUBTITLES="textstream"
https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/9378-DgQp6dfzKJ4WHMj-drm-aes-audio=256000-video=869702.m3u8?vbegin=8502&vend=8537
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=1718000,RESOLUTION=852x480,CODECS="mp4a.40.2,avc1.4D401E",AUDIO="audio-aacl-320",SUBTITLES="textstream"
https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/9378-DgQp6dfzKJ4WHMj-drm-aes-audio=320000-video=1299424.m3u8?vbegin=8502&vend=8537
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=2407000,RESOLUTION=1280x720,CODECS="mp4a.40.2,avc1.64001F",AUDIO="audio-aacl-320",SUBTITLES="textstream"
https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/9378-DgQp6dfzKJ4WHMj-drm-aes-audio=320000-video=1948875.m3u8?vbegin=8502&vend=8537
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=5638000,RESOLUTION=1920x1080,CODECS="mp4a.40.2,avc1.640028",AUDIO="audio-aacl-320",SUBTITLES="textstream"
https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/9378-DgQp6dfzKJ4WHMj-drm-aes-audio=320000-video=4997368.m3u8?vbegin=8502&vend=8537
#EXT-X-I-FRAME-STREAM-INF:BANDWIDTH=14000,RESOLUTION=256x144,CODECS="avc1.4D400C",URI="https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/keyframes/9378-DgQp6dfzKJ4WHMj-drm-aes-video=99963.m3u8?vbegin=8502&vend=8537"
#EXT-X-I-FRAME-STREAM-INF:BANDWIDTH=75000,RESOLUTION=426x240,CODECS="avc1.4D4015",URI="https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/keyframes/9378-DgQp6dfzKJ4WHMj-drm-aes-video=559836.m3u8?vbegin=8502&vend=8537"
#EXT-X-I-FRAME-STREAM-INF:BANDWIDTH=116000,RESOLUTION=640x360,CODECS="avc1.4D401E",URI="https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/keyframes/9378-DgQp6dfzKJ4WHMj-drm-aes-video=869702.m3u8?vbegin=8502&vend=8537"
#EXT-X-I-FRAME-STREAM-INF:BANDWIDTH=173000,RESOLUTION=852x480,CODECS="avc1.4D401E",URI="https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/keyframes/9378-DgQp6dfzKJ4WHMj-drm-aes-video=1299424.m3u8?vbegin=8502&vend=8537"
#EXT-X-I-FRAME-STREAM-INF:BANDWIDTH=259000,RESOLUTION=1280x720,CODECS="avc1.64001F",URI="https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/keyframes/9378-DgQp6dfzKJ4WHMj-drm-aes-video=1948875.m3u8?vbegin=8502&vend=8537"
#EXT-X-I-FRAME-STREAM-INF:BANDWIDTH=663000,RESOLUTION=1920x1080,CODECS="avc1.640028",URI="https://cdn.prod.medicitv.fr/9378/files/24/06/25/10225979/9378-DgQp6dfzKJ4WHMj-drm-aes.ism/keyframes/9378-DgQp6dfzKJ4WHMj-drm-aes-video=4997368.m3u8?vbegin=8502&vend=8537"
#EXT-X-SESSION-KEY:METHOD=AES-128,URI="https://sdn-global-prog-cache-aka.3qsdn.com/9378/files/24/06/25/10225979/Hj34LQp9N7RCwV6-9378-DgQp6dfzKJ4WHMj-drm-aes.ism.key"
```

That's what an HLS manifest is supposed to look like.  This one is a master manifest.  You can tell that from the multiple occurrences of lines beginning with `#EXT-X-STREAM-INF` with various video resolutions, each of which also mentions a URL that contains the string `m3u8`.  (You'll have to scroll that block horizontally to see the instances of `m3u8` in the URLs.)  Those URLs are the locators for the various stream manifests that describe this single video at the various resolutions.  There's also captions here.  I must point out that Medici TV is based in Paris, France.  Notice that there's no French in the manifest.

The point is that the Chinese web site is simply not following the rules.

**Comment by garpereira** [comment]
I'm experiencing the same issue. I believe it might be due to DRM protection enabled on certain video hosts. For example, it's working fine with YouTube and bakashi.tv videos, but on Crunchyroll, where DRM is enabled, the extension doesn't work properly.

**Reply by GrampaWildWilly** [reply]
Yes, DRM will cause VDH to fail.  Generally, it downloads something but it's encrypted.  Media players don't know how to decrypt the content so it ends up being useless.  This is the intended purpose of DRM.  I've said before that DRM is being abused around the web.  Only a class action lawsuit, similar to the one from 50 or so years ago that enabled the videocassette recorder industry, is going to correct this situation.

**Reply by GrampaWildWilly** [reply]
However, it's not clear whether DRM is involved in the Chinese web site we've been discussing.  It looks to me like the web site is so non-standard, VDH has no hope of working there.  So I think we're talking about 2 unrelated topics.
